File: codeforces/B_Amr_and_The_Large_Array.py
# Link - https://codeforces.com/contest/558/problem/B
# Time: O(n)

from collections import defaultdict

n = int(input())
a = list(map(int, input().split()))

M, counter = 0, defaultdict(int)
first, last = {}, {}

for i, x in enumerate(a):
    counter[x] += 1
    M = max(M, counter[x])
    if x not in first:
        first[x] = i
    last[x] = i

l, r = -1, -1
for x in a:
    if counter[x] == M:
        if l == -1 or last[x] - first[x] < r - l:
            l, r = first[x], last[x]

print(l + 1, r + 1)

"""

# find the max freq
m, counter = 0, defaultdict(int)
for x in a:
    counter[x] += 1
    m = max(m, counter[x])

counter.clear()

i, j = 0, 0
l, r = -1, -1

# sliding window
while j < n:
    counter[a[j]] += 1
    while counter[a[j]] >= m:
        if l == -1 or r - l > j - i:
            l, r = i, j
        counter[a[i]] -= 1
        i += 1
    j += 1

print(l + 1, r + 1)


"""

File: codeforces/B_Amr_and_The_Large_Array.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

const int M = 10e6 + 5;

int start[M] = {}, last[M], freq[M] = {};

int main()
{

    int n, mf = 0, val, l = 1, r = n;
    cin >> n;

    for (int i = 1; i <= n; ++i)
    {
        cin >> val;

        if (!start[val])
            start[val] = i;

        last[val] = i;

        freq[val]++;

        mf = max(mf, freq[val]);

        if (freq[i] >= mf)
        {
            if (r - l > last[i] - start[i])
            {
                l = start[i];
                r = last[i];
            }
        }
    }

    cout << l << " " << r << endl;
}

/*

const int M = 10e6 + 5;

int tFreq[M] = {}, sFreq[M] = {};

int main()
{
    int n, mf = 0;
    cin >> n;

    vector<int> a(n);
    for (int &x : a)
    {
        cin >> x;
        tFreq[x]++;
        mf = max(mf, tFreq[x]);
    }

    int l = 0, r = 0, al = 0, ar = n - 1;
    while (r < n)
    {
        sFreq[a[r]]++;
        while (sFreq[a[r]] >= mf)
        {
            if (ar - al > r - l)
            {
                ar = r;
                al = l;
            }
            sFreq[a[l]]--;
            l++;
        }

        r++;
    }

    cout << al + 1 << " " << ar + 1 << endl;
}
*/